bleak/__init__.py

```diff
@@ -204,6 +206,55 @@ def set_scanning_filter(self, **kwargs):
         )
         self._backend.set_scanning_filter(**kwargs)
 
+    def __aiter__(self) -> AsyncIterator[Tuple[BLEDevice, AdvertisementData]]:
+        self._backend.new_devices = asyncio.Queue()
+        return self
+
+    async def __anext__(self) -> Tuple[BLEDevice, AdvertisementData]:
+        return await self._backend.new_devices.get()
+
+    @overload
+    def devices(
+        self, timeout: Optional[float] = None, return_adv: Literal[False] = False
+    ) -> AsyncIterator[BLEDevice]:
+        ...
+
+    @overload
+    def devices(
+        self, timeout: Optional[float] = None, return_adv: Literal[True] = True
+    ) -> AsyncIterator[Tuple[BLEDevice, AdvertisementData]]:
+        ...
+
+    async def devices(self, timeout: Optional[float] = None, return_adv: bool = False):
+        """
+        Continuously scan for devices and yield them.
+
+        Args:
+            timeout:
```

Comment:
I would rather leave out the timeout argument and let users wrap with `with asyncio.timeout(...):` if needed.